hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -250,29 +264,66 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
             return null;
           });
 
-      LOG.info("Renamed table from {}, to {}", from, to);
+      LOG.info("Renamed {} from {}, to {}", contentType.name(), fromIdentifier, to);
 
     } catch (NoSuchObjectException e) {
-      throw new NoSuchTableException("Table does not exist: %s", from);
+      switch (contentType) {
+        case TABLE:
+          throw new NoSuchTableException(
+              "Cannot rename %s to %s. Table does not exist", fromIdentifier, to);
+        case VIEW:
+          throw new NoSuchViewException(
+              "Cannot rename %s to %s. View does not exist", fromIdentifier, to);
+      }
 
     } catch (InvalidOperationException e) {
       if (e.getMessage() != null
           && e.getMessage().contains(String.format("new table %s already exists", to))) {
-        throw new org.apache.iceberg.exceptions.AlreadyExistsException(
-            "Table already exists: %s", to);
+        throwErrorForExistedToContent(fromIdentifier, removeCatalogName(toIdentifier));
```

Comment:
I don't think we want to do this here, as this would do another call to retrieve the table. I would leave the code here as it was before

Reply:
Motivation behind this changes is  
`ViewCatalogTests` expects error message like `Cannot rename %s to %s. %s already exists`. It should explicitly tell the upstream **TO** tableType. 

Also we are making only one call to retrieve TO content. 
Nevertheless, What do you propose to match the error messages for  `ViewCatalogTests` ?

Reply:
what I mean is that we shouldn't be doing whather it's doing in `throwErrorForExistedToContent()` at this point and just throw the `AlreadyExistsException` as it was done previously. We don't want do add new functionality in this PR IMO